Week3/homework/step2-3.js

```diff
@@ -30,9 +33,12 @@ console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
+  let i = 0;
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  do {
+    result = num <= 0 ? result : result + str;
```

Comment:
You are correct in that you need to test the value of `num` as a `do-while` loop only tests the loop condition at the end of the loop. However, since the value of `num` does not change inside the loop (it is 'loop invariant') it is better and more efficient to test the value of `num` once only before entering the loop and execute the loop only if `num > 0`:

```js
if (num > 0) {
  let i = 0;
  do {
    result += str;
    i++;
  } while (i < num);
}
```